src/views/settings/storage.ts

```diff
@@ -1,28 +1,20 @@
 import { homeDir } from "@tauri-apps/api/path";
 import { type as osType } from "@tauri-apps/api/os";
-
-export const DEFAULT_STORAGE_DIRS = {
-  Windows_NT: () =>
-    "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Team Fortress 2\\tf\\demos",
-  Darwin: (userHome: string) =>
-    `${userHome}/Library/Application Support/Steam/steamapps/common/Team Fortress 2/tf/demos`,
-  Linux: (userHome: string) =>
-    `${userHome}/.local/share/Steam/steamapps/common/Team Fortress 2/tf/demos`,
-};
+import { getTf2Dir } from "../../api";
 
 /**
  * Gets the default platform-specific demo file storage directory.
  * Only supports Windows, Mac, and Linux.  This will return `null` if run on any other platform.
  */
-export async function getPlatformStorageDir(): Promise<string | null> {
+export async function getDefaultDemosDir(): Promise<string | null> {
   const type = await osType();
+  const separator = type === "Windows_NT" ? "\\" : "/";
   const userHomeDir = await homeDir();
 
-  if (Object.keys(DEFAULT_STORAGE_DIRS).includes(type)) {
-    // eg setDefaultDir(DEFAULT_DIRS["Linux"]("/home/me/"))
-    // regex replace to normalize multiple consecutive slashes (/home/me///.local/ becomes /home/me/.local/)
-    return DEFAULT_STORAGE_DIRS[type](userHomeDir).replace(/\/{2,}/, "/");
-  } else {
-    return null;
-  }
+  // Try getting the TF2 game directory, falling back to the user home directory if it couldn't be found
+  const tf2Dir = await getTf2Dir().catch((e) => userHomeDir);
+
+  return [tf2Dir, "tf", "demos"]
+    .join(separator)
+    .replace(/[\\/]{2,}/, separator); // in case userHomeDir has a trailing slash
```

Comment:
You can delegate all of that platform-specific code to tauri.
See https://tauri.app/v1/api/js/path#join.